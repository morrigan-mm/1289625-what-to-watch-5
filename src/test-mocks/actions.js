export const unknownAction = {type: `NOOP`};
