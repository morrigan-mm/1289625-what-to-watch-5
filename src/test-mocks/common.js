export const mockEvent = {
  preventDefault: () => {}
};

export const noop = () => {};
