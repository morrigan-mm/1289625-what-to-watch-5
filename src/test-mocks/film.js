const film = {
  id: 15,
  poster: `#poster`,
  preview: `#preview`,
  video: `#video`,
  cover: `#cover`,
  title: `The Revenant`,
  genre: `Action`,
  releaseDate: 2015,
  description: `A frontiersman on a fur trading`,
  rating: 4,
  ratingCount: 618498,
  director: `The Director`,
  actors: [`actor1`, `actor2`],
  runtime: 156,
  addedToMyList: false,
  backgroundColor: `#92918B`,
  backgroundImage: `#bgimage`
};

export default film;
