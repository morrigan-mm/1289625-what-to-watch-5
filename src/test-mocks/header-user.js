import {AuthorizationStatus} from "../constants";

export const headerUserAuthorized = {
  authorizationStatus: AuthorizationStatus.AUTH,
  avatar: `#`
};
export const headerUserUnauthorized = {
  authorizationStatus: AuthorizationStatus.NO_AUTH
};
