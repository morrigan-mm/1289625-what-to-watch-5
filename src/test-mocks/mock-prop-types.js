import PropTypes from "prop-types";

export const mockComponentType = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};
