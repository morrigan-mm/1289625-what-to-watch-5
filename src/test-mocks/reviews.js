const reviews = [
  {
    id: 1,
    text: `some review text 1`,
    rating: 1.1,
    author: `Author name1`,
    reviewDate: `2020-01-01T14:13:56.569Z`
  },
  {
    id: 2,
    text: `some review text 2`,
    rating: 2.2,
    author: `Author name2`,
    reviewDate: `2020-02-02T14:13:56.569Z`
  },
  {
    id: 3,
    text: `some review text 3`,
    rating: 3.3,
    author: `Author name3`,
    reviewDate: `2020-03-03T14:13:56.569Z`
  }
];

export default reviews;
