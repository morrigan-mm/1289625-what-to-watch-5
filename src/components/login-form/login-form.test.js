import React from "react";
import renderer from "react-test-renderer";
import {noop} from "../../test-mocks/common";
import LoginForm from "./login-form";

describe(`LoginForm`, () => {
  it(`Should render correctly`, () => {
    const tree = renderer
      .create(
          <LoginForm onSubmit={noop} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should render authorization error message`, () => {
    const tree = renderer
      .create(
          <LoginForm onSubmit={noop} authorizationError={401} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
