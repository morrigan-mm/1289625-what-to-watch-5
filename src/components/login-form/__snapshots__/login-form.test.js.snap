// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Should render authorization error message 1`] = `
<form
  action="#"
  className="sign-in__form"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="sign-in__message"
  >
    <p>
      We canâ€™t recognize this email and password combination. Please try again.
    </p>
  </div>
  <div
    className="sign-in__fields"
  >
    <div
      className="sign-in__field"
    >
      <input
        className="sign-in__input"
        id="user-email"
        name="user-email"
        onChange={[Function]}
        placeholder="Email address"
        type="email"
        value=""
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-email"
      >
        Email address
      </label>
    </div>
    <div
      className="sign-in__field"
    >
      <input
        className="sign-in__input"
        id="user-password"
        name="user-password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-password"
      >
        Password
      </label>
    </div>
  </div>
  <div
    className="sign-in__submit"
  >
    <button
      className="sign-in__btn"
      type="submit"
    >
      Sign in
    </button>
  </div>
</form>
`;

exports[`LoginForm Should render correctly 1`] = `
<form
  action="#"
  className="sign-in__form"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="sign-in__fields"
  >
    <div
      className="sign-in__field"
    >
      <input
        className="sign-in__input"
        id="user-email"
        name="user-email"
        onChange={[Function]}
        placeholder="Email address"
        type="email"
        value=""
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-email"
      >
        Email address
      </label>
    </div>
    <div
      className="sign-in__field"
    >
      <input
        className="sign-in__input"
        id="user-password"
        name="user-password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-password"
      >
        Password
      </label>
    </div>
  </div>
  <div
    className="sign-in__submit"
  >
    <button
      className="sign-in__btn"
      type="submit"
    >
      Sign in
    </button>
  </div>
</form>
`;
